File: python/leetcode/67-add-binary.py
from collections import deque


class Solution:
    def addBinary(self, a: str, b: str) -> str:

        digits_a = [int(ch) for ch in a]
        digits_b = [int(ch) for ch in b]

        carry = 0
        res = deque()

        while digits_a or digits_b:
            result = carry
            if digits_a:
                result += digits_a.pop()
            if digits_b:
                result += digits_b.pop()

            res.appendleft(result % 2)
            carry = result // 2

        if carry:
            res.appendleft(carry)

        return "".join([str(x) for x in list(res)])

File: c#/leetcode/67-add-binary.cs
public class Solution
{
  public string AddBinary(string a, string b)
  {
    char[] digitA = a.ToCharArray();
    char[] digitB = b.ToCharArray();

    int carry = 0;
    int i = digitA.Length - 1;
    int j = digitB.Length - 1;

    var res = new List<int>();

    while (i >= 0 || j >= 0)
    {
      int result = carry;
      if (i >= 0)
      {
        result += (int)char.GetNumericValue(digitA[i]);
        i--;
      }
      if (j >= 0)
      {
        result += (int)char.GetNumericValue(digitB[j]);
        j--;
      }
      res.Add(result % 2);
      carry = result / 2;

    }

    while (carry > 0)
    {
      res.Add(carry % 2);
      carry = carry / 2;
    }

    res.Reverse();



    return string.Join("", res);

  }
}